<launch>
  <node pkg="color_blob_detector" type="multiple_hue_detector_monitor" name="$(anon multiple_hue)" output="screen" clear_params="true">

    <!-- generic parameters for perceptual node -->

    <param name="update_rate" value="1" type="int"/> 
    <param name="display_image" value="false" type="bool"/> 
    <param name="publish_markers" value="true" type="bool"/> 

    <param name="base_frame" value="/baseLink" type="str"/> 

    <param name="source_identifier" value="/kinect/camera/rgb/filtered_points" type="str"/>
    <param name="output_identifier" value="/kinect/camera/rgb/feature/color/hue" type="str"/>
    <param name="marker_identifier" value="/kinect/camera/rgb/feature/color/hue/marker" type="str"/>

    <param name="feature_type" value="color" type="str"/> 
    <param name="feature_name" value="hue" type="str"/> 

    <!-- other variations on hue detector, possible attributes -->

    <param name="blur_kernel" value="1" type="int"/> 
    <param name="max_cluster_size" value="2000" type="int"/> 
    <param name="min_cluster_size" value="500" type="int"/>
    <param name="within_cluster_distance" value=".01" type="double"/>
    <param name="max_num_features" value="1" type="int"/>  <!-- max num features for each hue, should be small -->

    <!-- attributes -->
    
    <rosparam param="attribute_names">['hue']</rosparam>

    <rosparam param="attribute_mins">
      hue: [0.0]
    </rosparam>
    <rosparam param="attribute_maxes">
      hue: [179.0]
    </rosparam> 
    <param name="display_hue" value="170" type="int"/> 
    
    <rosparam param="attribute_windows">
      hue: [10.0]
    </rosparam>
    <rosparam param="attribute_steps">
      hue: [1.0]
    </rosparam>


  </node>
</launch>

